src/__tests__/datasource.test.ts

```diff
@@ -174,10 +182,7 @@ describe('Datasource Query', () => {
       expect(result).toEqual({ data: [] });
     });
     it('should display errors for malformed queries', () => {
-      expect(tsTargetA.error).toBeDefined();
-      expect(tsTargetA.error).not.toHaveLength(0);
-      expect(tsTargetB.error).toBeDefined();
-      expect(tsTargetB.error).not.toHaveLength(0);
+      expect(appEvents.emit).toHaveBeenCalledTimes(2);
```

Comment:
maybe better to check the exact error objects it has been called with?

Reply:
they are pretty complex, but I agree that we need to check error messages itself